src/Core/Services/EnvironmentService.cs

```diff
@@ -96,15 +110,57 @@ public async Task SetUrlsFromStorageAsync()
 
         }
 
+        private async Task<Region?> MigrateToRegionsAsync(EnvironmentUrlData urls)
+        {
+            if (urls.Base == Region.US.BaseUrl())
+            {
+                await _stateService.UpdateActiveUserEnvironmentAsync(Region.US, Region.US.GetUrls());
+                return Region.US;
+            }
+            if (urls.Base == Region.EU.BaseUrl())
+            {
+                await _stateService.UpdateActiveUserEnvironmentAsync(Region.EU, Region.EU.GetUrls());
+                return Region.EU;
+            }
+            await _stateService.UpdateActiveUserEnvironmentAsync(Region.SelfHosted, urls);
+            return Region.SelfHosted;
```

Comment:
Added extra check to avoid settings urls to null if environment is SelfHosted.
`await _stateService.UpdateActiveUserEnvironmentAsync(region, region.GetUrls() ?? urls);`